pkgs/by-name/in/install-grub-ng/src/builder.rs

```diff
@@ -0,0 +1,153 @@
+mod appearance;
+mod entries;
+mod install;
+
+use std::{
+    collections::HashSet,
+    fmt::Write as _,
+    fs,
+    os::{linux::fs::MetadataExt, unix::fs::PermissionsExt},
+    path::{Path, PathBuf},
+};
+
+use eyre::Result;
+
+use crate::{
+    config::{Config, Password},
+    grub::Grub,
+};
+
+pub struct Builder<'conf> {
+    config: Config<'conf>,
+
+    grub_boot: Grub,
+    grub_store: Option<Grub>,
+
+    default_config: &'conf Path,
+    pub copied: HashSet<PathBuf>,
+
+    dry_run: bool,
+
+    inner: String,
+}
+impl<'conf> Builder<'conf> {
+    pub fn new(mut config: Config<'conf>, default_config: &'conf Path) -> Result<Self> {
+        let dry_run = std::env::var("DRY_RUN").as_deref() == Ok("true");
+
+        if !dry_run {
+            let grub = config.boot_path.join("grub");
+            fs::create_dir_all(&grub)?;
+            fs::set_permissions(&grub, fs::Permissions::from_mode(0o700))?;
```

Comment:
Any parent dirs will have unexpected permissions here.

Reply:
If `create_dir_all` creates any parent directories, they won't use the permission we set just below.

I'm not sure what's the best thing to do here.